c++/src/kj/compat/http.c++

```diff
@@ -2339,10 +2336,13 @@ private:
   uint64_t length;
 
   kj::Promise<void> maybeFinishAfter(kj::Promise<void> promise) {
-    if (length == 0) {
-      return promise.then([this]() { doneWriting(); });
-    } else {
-      return kj::mv(promise);
```

Comment:
The old code here was a no-op in the common case that we're not at the end of the stream yet. Only in the very last call, once per stream, it would allocate a `.then()`.

The new code, unfortunately, allocates a coroutine for every write.

One way to approximate the old performance would be to factor out just `promise.then([this]() { doneWriting(); })` into a separate coroutine. `maybeFinishAfter()` itself would not need to be a coroutine, it just conditionally calls the other coroutine.

However, that would still be more verbose than the old code, defeating the purpose of the coroutine conversion. Maybe we could just leave this one as-is?

Reply:
Yeah, very good points. I've reverted this part.